backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java

```diff
@@ -1277,11 +1357,24 @@ public void deleteProjects(
         throw new InvalidArgumentException("Project IDs not found in DeleteProjects request");
       }
 
+      List<GetResourcesResponseItem> responseItems =
+          roleService.getResourceItems(
+              null, new HashSet<>(request.getIdsList()), ModelDBServiceResourceTypes.PROJECT);
       List<String> deletedProjectIds = projectDAO.deleteProjects(request.getIdsList());
-      saveAuditLogs(null, ModelDBConstants.DELETE, deletedProjectIds, "");
-
-      responseObserver.onNext(
-          DeleteProjects.Response.newBuilder().setStatus(!deletedProjectIds.isEmpty()).build());
+      DeleteProjects.Response response =
+          DeleteProjects.Response.newBuilder().setStatus(!deletedProjectIds.isEmpty()).build();
+      Long workspaceId =
+          responseItems.stream().findFirst().isPresent()
+              ? responseItems.stream().findFirst().get().getWorkspaceId()
+              : null;
```

Comment:
that is only for the first of projects from the list, right? See https://github.com/VertaAI/modeldb/pull/1980, maybe it's better

Reply:
I don't understand what you mean

Reply:
I mean that deleted projects may have different workspace ids. Personal workspace can be used here?

Reply:
The new protos should resolve this, right?

Reply:
I think so. We decided to use multiple workspaces?

Reply:
We have logging multiple resourceIds at once in single audit log entry. and we do not have provision for now to use endpoint with multiple workspaceId's. So workspaceId will always there.

Reply:
This change has multiple workspace ids: https://github.com/VertaAI/protos-all/pull/162/files Is it a mistake?

Reply:
PR: https://github.com/VertaAI/modeldb/pull/2027 will supporting resource and workspace together in single entry.

Reply:
Resolved in slack